### PR TITLE
feat: Print operator stats in query replayer

### PR DESCRIPTION
**Pull request by duanmeng**
Print operator stats in query replayer.

Part of #9668

### CODE REVIEW

File: velox/tool/trace/OperatorReplayerBase.cpp [hunk 1]
```diff
@@ -103,4 +108,15 @@ OperatorReplayerBase::replayNodeFactory(const core::PlanNode* node) const {
     return createPlanNode(node, nodeId, source);
   };
 }
+
+void OperatorReplayerBase::printOperatorStats(
```
Comment (xiaoxmeng) [inline velox/tool/trace/OperatorReplayerBase.cpp:112]:
```
s/printOperatorStats/printStats/
```
Could you also print out the tree memory usage by calling treeMemoryUsage without skipping empty pool? thanks!

File: velox/tool/trace/TableScanReplayer.cpp [hunk 2]
```diff
@@ -28,13 +28,17 @@ using namespace facebook::velox::exec::test;
...
 RowVectorPtr TableScanReplayer::run() {
+  std::shared_ptr<exec::Task> task;
   const auto plan = createPlan();
-  return exec::test::AssertQueryBuilder(plan)
-      .maxDrivers(driverIds_.size())
-      .configs(queryConfigs_)
-      .connectorSessionProperties(connectorConfigs_)
-      .splits(getSplits())
-      .copyResults(memory::MemoryManager::getInstance()->tracePool());
+  const auto result =
```
Comment (xiaoxmeng) [inline velox/tool/trace/TableScanReplayer.cpp:33]: You can use capaturePlanNodeId to store the replay operator id as a member of the base class. Thanks!

**Review by xiaoxmeng — COMMENTED** [review]
@duanmeng thanks for the improvement!

**Review by xiaoxmeng — APPROVED** [review]
@duanmeng LGTM. Thanks for the update!

File: velox/tool/trace/OperatorReplayerBase.cpp [hunk 3]
```diff
@@ -103,4 +110,15 @@ OperatorReplayerBase::replayNodeFactory(const core::PlanNode* node) const {
...
+void OperatorReplayerBase::printStats(
+    const std::shared_ptr<exec::Task>& task) const {
+  const auto planStats = exec::toPlanStats(task->taskStats());
+  const auto& stats = planStats.at(replayPlanNodeId_);
+  for (const auto& [name, operatorStats] : stats.operatorStats) {
+    LOG(INFO) << "Stats of replaying operator " << name << " : "
+              << operatorStats->toString();
+  }
+  LOG(INFO) << "Memory usage: " << task->pool()->treeMemoryUsage(false);
+}
```
Comment (tanjialiang) [inline velox/tool/trace/OperatorReplayerBase.cpp:123]: This does not need to be a member of OperatorReplayerBase if it is not using any members of this class. We can make it static or simply put it as a helper outside of the class.

File: velox/tool/trace/PartitionedOutputReplayer.cpp [hunk 4]
```diff
@@ -150,6 +150,7 @@ RowVectorPtr PartitionedOutputReplayer::run() {
       executor_.get(),
       consumerExecutor_.get(),
       consumerCb_);
+  printStats(task);
```
Comment (tanjialiang) [inline velox/tool/trace/PartitionedOutputReplayer.cpp:153]: This should be a common thing. Should we abstract it out to put it in the framework in some way? Does not seem to be clean to make individual calls in every single impl. Newly added impls might miss this printing.

File: velox/tool/trace/OperatorReplayerBase.cpp (hunk 3 above)
Comment (duanmeng) [reply velox/tool/trace/OperatorReplayerBase.cpp:123]: Yes, it uses `OperatorReplayerBase ::replayPlanNodeId_`

**Pull request closed without merging**